test/test_page_without_a_file.rb

```diff
@@ -0,0 +1,123 @@
+# frozen_string_literal: true
+
+require "helper"
+
+class TestPageWithoutAFile < JekyllUnitTest
+  def setup_page(*args)
+    dir, file = args
+    if file.nil?
+      file = dir
+      dir = ""
+    end
+    @page = PageWithoutAFile.new(@site, source_dir, dir, file)
+  end
+
+  context "A PageWithoutAFile" do
+    setup do
+      clear_dest
+      @site = Site.new(Jekyll.configuration({
+        "source"            => source_dir,
+        "destination"       => dest_dir,
+        "skip_config_files" => true,
+      }))
+    end
+
+    context "on intialized" do
```

Comment:
`on intialize` ? Or maybe even `with default config` ? What do you think?

Reply:
I disagree with `with default config`.. `PageWithoutAFile` instances do not change on passing different parameters. For a given site, its always `nil` content and `empty` data hash.